optaplanner-core/src/main/java/org/optaplanner/core/impl/score/stream/common/AbstractConstraintSessionFactory.java

```diff
@@ -0,0 +1,61 @@
+/*
+ * Copyright 2020 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.optaplanner.core.impl.score.stream.common;
+
+import java.math.BigDecimal;
+import java.util.Arrays;
+
+import org.optaplanner.core.api.score.Score;
+import org.optaplanner.core.impl.domain.solution.descriptor.SolutionDescriptor;
+import org.optaplanner.core.impl.score.definition.ScoreDefinition;
+import org.optaplanner.core.impl.score.stream.ConstraintSessionFactory;
+
+public abstract class AbstractConstraintSessionFactory<Solution_> implements ConstraintSessionFactory<Solution_> {
+
+    private final ScoreDefinition scoreDefinition;
```

Comment:
Raw type.

Reply:
It's raw on SolutionDescriptor too :(
I am really looking for to 8.0 to get reduce Score<S extends Score> to just Score,
that will give a fighting change against the raw types.

Reply:
Ps: I don't recommending fixing SolutionDescriptor for this right now. I tried in the past and failed, eventually - we need to degenerify Score first.